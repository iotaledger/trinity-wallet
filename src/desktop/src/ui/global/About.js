/* global Electron */
import React from 'react';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import ReactMarkdown from 'react-markdown';

import Modal from 'ui/components/modal/Modal';
import Button from 'ui/components/Button';
import Scrollbar from 'ui/components/Scrollbar';
import Icon from 'ui/components/Icon';

import settings from '../../../package.json';

import changelog from '../../../CHANGELOG.md';

import css from './about.scss';

/**
 * About window component
 */
export class AboutComponent extends React.PureComponent {
    static propTypes = {
        /** @ignore */
        t: PropTypes.func.isRequired,
    };

    state = {
        visible: false,
    };

    componentDidMount() {
        this.onMenuToggle = this.menuToggle.bind(this);
        Electron.onEvent('menu', this.onMenuToggle);
    }

    componentWillUnmount() {
        Electron.removeEvent('menu', this.onMenuToggle);
    }

    /**
     * Proxy native menu triggers to an action
     * @param {string} Item - Triggered menu item
     */
    menuToggle(item) {
        this.setState({
            visible: item === 'about',
        });
    }

    render() {
        const { visible } = this.state;
        const { t } = this.props;

        return (
            <Modal variant="global" isOpen={visible} onClose={() => this.setState({ visible: false })}>
                <section className={css.about}>
                    <Icon icon="iota" size={48} />
                    <h1>Trinity Wallet</h1>
                    <h2>v{settings.version}</h2>

                    <article>
                        <Scrollbar>
                            <ReactMarkdown source={changelog} />
                        </Scrollbar>
                    </article>
                    <footer>
                        <Button onClick={() => this.setState({ visible: false })} variant="dark">
                            {t('close')}
                        </Button>
                    </footer>
                </section>
            </Modal>
        );
    }
}

export default withTranslation()(AboutComponent);
