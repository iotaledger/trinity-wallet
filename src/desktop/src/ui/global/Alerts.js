import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import withAlertsData from 'containers/global/Alerts';

import Icon from 'ui/components/Icon';

import css from './alerts.scss';

/**
 * Alerts UI helper component
 */
class Alerts extends React.PureComponent {
    static timeout = null;

    static propTypes = {
        /** @ignore */
        disposeOffAlert: PropTypes.func.isRequired,
        /** @ignore */
        alerts: PropTypes.object.isRequired,
    };

    componentWillReceiveProps(nextProps) {
        if (this.timeout) {
            clearTimeout(this.timeout);
        }
        if (nextProps.alerts.category.length && nextProps.alerts.closeInterval > 0) {
            this.timeout = setTimeout(() => {
                this.props.disposeOffAlert();
            }, nextProps.alerts.closeInterval);
        }
    }

    closeAlert() {
        if (this.timeout) {
            clearTimeout(this.timeout);
        }
        this.props.disposeOffAlert();
    }

    render() {
        const { alerts, disposeOffAlert } = this.props;

        return (
            <div className={css.wrapper}>
                <div
                    onClick={() => disposeOffAlert()}
                    className={classNames(alerts.category.length ? css.visible : null, css[`${alerts.category}`])}
                >
                    <span>
                        <Icon icon="cross" size={14} />
                    </span>
                    {alerts.title && <h2>{alerts.title}</h2>}
                    {alerts.message && <p>{alerts.message}</p>}
                </div>
            </div>
        );
    }
}

export default withAlertsData(Alerts);
