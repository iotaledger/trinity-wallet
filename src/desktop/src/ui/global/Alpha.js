import React from 'react';

import Modal from 'ui/components/modal/Modal';
import Button from 'ui/components/Button';

/**
 * Alpha release warning modal
 */
class AlphaWarning extends React.PureComponent {
    state = {
        active: true,
    };

    render() {
        const { active } = this.state;

        return (
            <Modal variant="global" isOpen={active} onClose={() => {}}>
                <section>
                    <article>
                        <h1 style={{ color: 'var(--negative)' }}>Alpha release warning</h1>
                        <p>
                            This is an early (alpha) release of the wallet. It is to be used for testing purposes only,
                            with very small amounts of tokens.
                        </p>
                    </article>
                    <footer>
                        <Button onClick={() => this.setState({ active: false })} variant="dark">
                            Continue
                        </Button>
                    </footer>
                </section>
            </Modal>
        );
    }
}

export default AlphaWarning;
