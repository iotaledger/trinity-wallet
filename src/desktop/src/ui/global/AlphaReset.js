/*global Electron*/
import React from 'react';

import { DESKTOP_VERSION } from 'config';

import Button from 'ui/components/Button';

import css from 'ui/views/onboarding/index.scss';

/**
 * Alpha version activation view
 */
class Activation extends React.PureComponent {
    resetWallet = (e) => {
        e.preventDefault();

        localStorage.clear();

        Electron.clearStorage();
        Electron.setActiveVersion(DESKTOP_VERSION);

        location.reload();
    };

    render() {
        return (
            <main className={css.onboarding}>
                <header />
                <section>
                    <form className="center" onSubmit={this.resetWallet}>
                        <fieldset>
                            <h2>This alpha release requires a full wallet reset!</h2>
                            <p>The wallet will be reset to defaults and all your saved data will be lost.</p>
                            <Button type="submit" variant="primary">
                                Reset wallet
                            </Button>
                        </fieldset>
                    </form>
                </section>
                <footer />
            </main>
        );
    }
}

export default Activation;
