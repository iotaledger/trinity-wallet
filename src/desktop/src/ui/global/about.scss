.about {
    article {
        display: block;
        position: relative;
        max-width: 680px;
        max-height: 300px;
        width: calc(100% - 20px);
        height: calc(100vh - 200px);
        margin: 40px auto;
        text-align: left;
        background: var(--box-bg);
        padding: 20px 10px 20px 20px;
    }

    h1 {
        margin: 20px auto;
        line-height: 18px;
        font-size: 38px;
    }

    p {
        font-size: 14px;
        strong {
            color: var(--negative);
        }
    }

    h2 {
        display: inline-block;
        color: var(--primary);
        font-size: 18px;
        margin-bottom: 0px;

        small {
            margin: 5px 0 0 6px;
            float: right;
            font-size: 11px;
            padding: 0 10px 1px;
            border-radius: 3px;
            background: #c65b38;
            color: #fff;
        }
    }

    h5 {
        font-size: 18px;
        font-weight: 400;
        margin-bottom: 8px;
    }
    ul {
        margin-bottom: 20px;
    }

    li {
        font-size: 14px;
        line-height: 21px;
        margin: 0 0 4px 24px;
        list-style: circle;
    }
}
