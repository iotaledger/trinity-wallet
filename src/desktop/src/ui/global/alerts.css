.wrapper {
    position: fixed;
    right: 0;
    top: 0;
    z-index: 1002;

    > div {
        @media print {
            display: none;
        }

        position: relative;
        margin: 8px;
        padding: 8px 16px 16px 24px;
        width: 320px;
        background-color: var(--body-bg);
        border: none;
        max-width: calc(100% - 32px);

        h2 {
            margin: 0;
            margin-bottom: 4px;
        }

        span {
            cursor: pointer;
            position: absolute;
            top: 0px;
            right: 0px;
            font-size: 16px;
            line-height: 1;
            padding: 12px;
        }

        &.info {
            background: var(--positive);
            color: var(--positive-body);
        }

        &.error {
            background: var(--negative);
            color: var(--negative-body);
        }

        &.success {
            background: var(--positive);
            color: var(--positive-body);
        }

        &.extra {
            background: var(--extra);
            color: var(--extra-body);
        }

        &.warning {
            background: var(--negative);
            color: var(--negative-body);
        }
    }
}
