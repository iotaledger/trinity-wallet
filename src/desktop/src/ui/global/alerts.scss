.wrapper {
    position: fixed;
    right: 0;
    top: 0;
    z-index: 1002;
    pointer-events: none;

    > div {
        @media print {
            display: none;
        }

        display: block;
        position: relative;
        margin: 50px 20px 0;
        padding: 18px 32px 24px 24px;
        min-width: 340px;
        width: 320px;
        background: var(--body-bg);
        max-width: calc(100% - 32px);
        min-height: 120px;
        transform: scale(0.8);
        opacity: 0;
        pointer-events: none;
        border-radius: 6px;

        * {
            cursor: pointer;
        }

        h2 {
            margin: 0;
            font-size: 21px;
            margin-bottom: 16px;
            font-weight: 400;
        }

        p {
            font-size: 16px;
            line-height: 24px;
            margin-bottom: 0px;
            word-break: break-word;
        }

        span {
            cursor: pointer;
            position: absolute;
            top: 6px;
            right: 6px;
        }

        &.visible {
            opacity: 1;
            pointer-events: all;
            cursor: pointer;
            transform: scale(1);
            transition: opacity 0.12s ease-in, transform 0.24s cubic-bezier(0.23, 1, 0.32, 1);
        }

        &.info {
            background: var(--positive);
            color: var(--positive-body);
            border-color: var(--positive-hover);
            span:hover {
                color: var(--positive-hover);
            }
        }

        &.error {
            background: var(--negative);
            color: var(--negative-body);
            border-color: var(--negative-hover);
            span:hover {
                color: var(--negative-hover);
            }
        }

        &.success {
            background: var(--positive);
            color: var(--positive-body);
            border-color: var(--positive-hover);
            span:hover {
                color: var(--positive-hover);
            }
        }
    }
}
