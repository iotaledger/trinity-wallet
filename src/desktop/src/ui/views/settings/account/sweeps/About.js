import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withTranslation } from 'react-i18next';

import Button from 'ui/components/Button';
import Info from 'ui/components/Info';

import css from './index.scss';

/**
 * Sweep funds about component
 */
class About extends React.PureComponent {
    static propTypes = {
        /** @ignore */
        history: PropTypes.shape({
            goBack: PropTypes.func.isRequired,
            push: PropTypes.func.isRequired,
        }).isRequired,
        /** @ignore */
        t: PropTypes.func.isRequired,
    };

    render() {
        const { t } = this.props;

        return (
            <form>
                <section className={css.long}>
                    <h1>{t('sweeps:lockedFundsRecovery')}</h1>
                    <Info>
                        <p>{t('sweeps:lockedFundsRecoveryExplanation')}</p>
                        <p>{t('sweeps:lockedFundsToolExplanation')}</p>
                        <p>
                            <strong>{t('sweeps:lockedFundsOwnRiskExplanation')}</strong>
                        </p>
                    </Info>
                </section>
                <footer className={css.choiceDefault}>
                    <div>
                        <Button
                            id="to-cancel"
                            onClick={() => this.props.history.goBack()}
                            className="square"
                            variant="dark"
                        >
                            {t('global:cancel')}
                        </Button>
                        <Button
                            id="to-transfer-funds"
                            onClick={() => this.props.history.push('/sweeps/transfer-funds')}
                            className="square"
                            variant="primary"
                        >
                            {t('global:accept')}
                        </Button>
                    </div>
                </footer>
            </form>
        );
    }
}

const mapStateToProps = (state) => ({
    wallet: state.wallet,
    additionalAccountName: state.accounts.accountInfoDuringSetup.name,
});

export default connect(mapStateToProps)(withTranslation()(About));
