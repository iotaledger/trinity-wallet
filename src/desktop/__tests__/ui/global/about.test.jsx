import React from 'react';
import { shallow } from 'enzyme';

import { AboutComponent as About } from 'ui/global/About';

const props = {
    t: (str) => str,
};

global.Electron = {
    onEvent: function(_event, e) {
        this.menuEvent = e;
    },
    removeEvent: () => {},
    menuEvent: null,
};

describe('About component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<About {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Set visible on menu event', () => {
        const wrapper = shallow(<About {...props} />);

        Electron.menuEvent('about');
        expect(wrapper.state('visible')).toBeTruthy();
    });
});
