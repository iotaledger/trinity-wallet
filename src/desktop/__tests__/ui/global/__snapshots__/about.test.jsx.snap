// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component Render the component 1`] = `
<Modal
  isOpen={false}
  onClose={[Function]}
  variant="global"
>
  <section
    className="about"
  >
    <Icon
      icon="iota"
      size={48}
    />
    <h1>
      Trinity Wallet
    </h1>
    <h2>
      v
      1.0.0
    </h2>
    <article>
      <Scrollbar>
        <ReactMarkdown
          astPlugins={Array []}
          escapeHtml={true}
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          source="### Foo
- Bar
- Baz"
          sourcePos={false}
          transformLinkUri={[Function]}
        />
      </Scrollbar>
    </article>
    <footer>
      <Button
        onClick={[Function]}
        variant="dark"
      >
        close
      </Button>
    </footer>
  </section>
</Modal>
`;
